ingest/sqs/pools/transformer/alloy_transmuter.go

```diff
@@ -0,0 +1,100 @@
+package poolstransformer
+
+import (
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/osmosis/v25/ingest/sqs/domain"
+	"github.com/osmosis-labs/sqs/sqsdomain"
+)
+
+// updateAlloyTransmuterInfo updates cosmwasmPoolModel with alloyed transmuter specific info.
+// - It queries alloyed transmuter contract asset configs and share denom, the construct
+// `AlloyTransmuterData`. Share denom for alloyed transmuter is the alloyed denom.
+// - append the alloyed denom to pool denoms.
+func (pi *poolTransformer) updateAlloyTransmuterInfo(
+	ctx sdk.Context,
+	poolId uint64,
+	contractAddress sdk.AccAddress,
+	cosmWasmPoolModel *sqsdomain.CosmWasmPoolModel,
+	poolDenoms *[]string,
+) error {
+	assetConfigs, err := alloyTransmuterListAssetConfig(ctx, pi.wasmKeeper, poolId, contractAddress)
+	if err != nil {
+		return err
+	}
+
+	// share denom of alloy transmuter pool is an alloyed denom
+	alloyedDenom, err := alloyTransmuterGetShareDenom(ctx, pi.wasmKeeper, poolId, contractAddress)
+	if err != nil {
+		return err
+	}
+
+	// append alloyed denom to denoms
+	*poolDenoms = append(*poolDenoms, alloyedDenom)
+
+	cosmWasmPoolModel.Data.AlloyTransmuter = &sqsdomain.AlloyTransmuterData{
+		AlloyedDenom: alloyedDenom,
+		AssetConfigs: assetConfigs,
+	}
+
+	return nil
+}
+
+// alloyTransmuterListAssetConfig queries the asset configs of the alloyed transmuter contract.
+func alloyTransmuterListAssetConfig(
+	ctx sdk.Context,
+	wasmKeeper domain.WasmKeeper,
+	poolId uint64,
+	contractAddress sdk.AccAddress,
+) ([]sqsdomain.TransmuterAssetConfig, error) {
+	bz, err := wasmKeeper.QuerySmart(ctx, contractAddress, []byte(`{"list_asset_configs":{}}`))
+	if err != nil {
+		return nil, fmt.Errorf(
+			"error querying list_asset_configs for pool (%d) contrat_address (%s): %w",
+			poolId, contractAddress, err,
+		)
+	}
+	var assetConfigsResponse struct {
+		AssetConfigs []sqsdomain.TransmuterAssetConfig `json:"asset_configs"`
+	}
```

Comment:
nit: consider defining the struct at the top of the file

Reply:
I noticed that you already replied to this:
https://github.com/osmosis-labs/osmosis/pull/8291#discussion_r1628841399

Feel free to resolve any of the comments you disagree with. Let's not block on this